src/Features/CSharp/Portable/CodeFixes/Async/CSharpAddAwaitCodeFixProvider.cs

```diff
@@ -76,8 +72,19 @@ private bool IsCorrectReturnType(ExpressionSyntax expression, SemanticModel sema
 
         private static ExpressionSyntax ConvertToAwaitExpression(ExpressionSyntax expression)
         {
-            return SyntaxFactory.AwaitExpression(expression)
-                                .WithAdditionalAnnotations(Formatter.Annotation);
+            AwaitExpressionSyntax result;
+
+            if (expression.HasLeadingTrivia)
```

Comment:
You are right.  The tokens will be preserved which means we may want something like this:
``` C#
return SyntaxFactory.AwaitExpression(SyntaxFactory.ParenthesizedExpression(expression.WithoutTrivia()))
                                .WithTriviaFrom(expression)
                                .WithAdditionalAnnotations(Formatter.Annotation)
                                .WithAdditionalAnnotations(Simplifier.Annotation);
```

However we will need to add logic to detect the conditional case.  We should special case ConditionalExpressionSyntax and check to make sure both cases are convertable to Task.